vueui/src/components/view/machinery/vending.vue

```diff
@@ -0,0 +1,191 @@
+<template>
+  <div>
+    <template v-if="s.mode == 0 || s.sel_price == 0">
+      <div class="cancel-button">
+        <vui-input-search :input="products" v-model="output" :keys="['name']" autofocus :threshold="threshold" />
+        <vui-button :disabled="!s.coin" :params="{ remove_coin: 1 }" icon="sign-out-alt">{{ s.coin ? s.coin : "No coin inserted." }}</vui-button>
+      </div>
+      <div class="t-parent">
+        <vui-button v-for="vend in output" :key="vend.key" :class="vend.amount > 0 ? '' : 'no-stock'"
+                    class="t-child tooltip" :disabled="vend.amount == 0 || s.mode == 1" :params="{ vendItem: vend.key }">
+          <div class="t-container" :style="{ height: s.ui_size + 'px', width: s.ui_size + 'px'}">
+            <span :class="[vend.amount > 0 ? '' : 'no-stock', vend.icon_tag]" class="food-icon"/>
+            <span v-if="vend.price > 0" class="cart-icon fas ic-shopping-cart"/>
+            <span v-if="vend.price > 0" class="price">{{ vend.price }}电</span>
+            <span class="qty" :class="vend.amount > 0 ? '' : 'no-stock'">(x{{ vend.amount }})</span>
+          </div>
+          <span class="tooltiptext">{{ vend.name }}</span>
+        </vui-button>
+      </div>
+    </template>
+    <template v-else-if="s.sel_name && s.sel_price > 0">
+      <div class="t-parent">
+        <p>Item selected:<span class="purchase-icon" :class="s.sel_icon" />{{s.sel_name}}</p>
+        <p>Charge: {{s.sel_price}}电 / {{s.sel_price}}cr</p>
+        <p>Swipe your NanoTrasen ID or insert credits to purchase.</p>
+        <p v-if="s.message_err == 1" class="danger">{{s.message}}</p>
+        <div class="cancel-button">
+          <vui-button :params="{ cancelpurchase: 1 }" icon="undo">Cancel Transaction</vui-button>
+        </div>
+      </div>
+    </template>
+    <template v-else>
+      <vui-button class="cancel-button danger" :params="{ reset: 1}" icon="undo">Reset Machine</vui-button>
+    </template>
+  </div>
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      s: this.$root.$data.state,
+      output: [],
+      threshold: 0.3
+    }
+  },
+  computed: {
+    products() {
+      return Object.values(this.s.products);
+    }
+  }
+};
+</script>
+
+<style lang="scss" scoped>
+
+.t-parent > p {
+  text-align: center;
+  margin-top: 0px;
+  width: 100%;
+}
+
+.cancel-button {
+  width: 100%;
+  margin-bottom: 5px;
+}
+
+p.danger {
+  color: red;
+}
+
+.purchase-icon {
+  vertical-align: bottom;
+}
+
+.food-icon {
+  // zoom: 2;
+  transform: scale(2);
+  position: absolute;
+  left: 0;
+  right: 0;
+  margin-left: auto;
+  margin-right: auto;
+  margin-top: 12px;
+}
```

Comment:
Have you considered adding `image-rendering`?

```suggestion
.food-icon {
  // zoom: 2;
  transform: scale(2);
  -ms-interpolation-mode: nearest-neighbor;
  image-rendering: pixelated;
  image-rendering: -moz-crisp-edges;
  image-rendering: crisp-edges;
  position: absolute;
  left: 0;
  right: 0;
  margin-left: auto;
  margin-right: auto;
  margin-top: 12px;
}
```

In theory it should make the images much nicer. In practice it doesn't because Byond, but maybe at some point there'll be a new browser engine? At least it looks really cool when debugging the UI though...

![image](https://user-images.githubusercontent.com/781546/97751491-b5d56100-1af2-11eb-89bb-d4d76b96132e.png)

Reply:
i'll play around with it and see if maybe there's a version that'll work in BYOND

Reply:
I did try quite a bit and couldn't figure out anything, good luck.